### ISSUE TITLE
Figure out how to authenticate API requests

### ISSUE THREAD

**Issue by LeNPaul** [LeNPaul/physics-in-motion#56]
Currently it looks like the APIs are open to the world and can be called without any authentication needed.

**Comment by LeNPaul** [comment]
It seems that when requests are made from the browser with the user logged in, the request includes a `req.user.username` parameter that contains the logged in user. Should change all API endpoints to use this instead of having to pass in the username from the POST request. This means that the workaround that I did in order to extract the username from the current page to include in the POST request is no longer needed and should be removed.

**Comment by LeNPaul** [comment]
Will need to re-write the tests for the lessons API (#55). Once this is done, it will make sense to refactor endpoints so that GET or POST requests are used appropriately (#51)

**Comment by LeNPaul** [comment]
Need to also make sure that the API endpoints are truly authenticated (i.e. if I provide `req.user.username` in the request, will it still work?). How does the authentication truly work to make sure that requests are authenticated with PassportJS? Is there some kind of cookie that needs to be generated, or some kind of session ID?

**Comment by LeNPaul** [comment]
The requests that include a `req.user.username` parameter are requests that contain a session cookie initialized on the browser when a user logs in by PassportJs. These can be used to authenticate API requests that come from the browser, since if they are not included in the request, or if the user logs out (therefore expiring the session cookie), then the request will return a 500 HTTP code. 

This is only true if the API endpoint explicitly requires the `req.user.username` parameter in the back end code, which it should always do if we are working with user data on the backend. There might be a better way to do this (i.e. return a 401 code), but for now this is sufficient.
